objcgen/objcgenerator.cs

```diff
@@ -796,8 +796,124 @@ protected override void Generate (ProcessedMethod method)
 				GenerateDefaultValuesWrappers (objcsig, method.Method);
 		}
 
+		void ReturnArrayValue (Type t)
+		{
+			var typecode = Type.GetTypeCode (t);
+			implementation.WriteLine ("MonoArray* __resarr = (MonoArray *) __result;");
+			implementation.WriteLine ("int __resarrlength = mono_array_length (__resarr);");
+
+			if (typecode == TypeCode.Byte)
+				implementation.WriteLine ("NSMutableData* __resobj = [NSMutableData dataWithCapacity:__resarrlength];");
```

Comment:
It would be best to return a non-mutable `NSData` as is match what the managed code is doing, e.g.

```
NSData *__resobj = [NSData dataWithBytes:mono_array_addr (__resarr) length: __resarrlength];
```

That would also avoid the loop (see my other comment).